src/sat.cc

```diff
@@ -1702,19 +1707,13 @@ void Sat::JoinThreads() {
        map_it != workers_map_.end(); ++map_it) {
     for (WorkerVector::const_iterator it = map_it->second->begin();
          it != map_it->second->end(); ++it) {
-      logprintf(12, "Log: Reaping thread status %d\n", (*it)->ThreadID());
-      if ((*it)->GetStatus() != 1) {
-        logprintf(0,
-                  "Process Error: Thread %d failed with status %d at "
-                  "%.2f seconds\n",
-                  (*it)->ThreadID(), (*it)->GetStatus(),
-                  (*it)->GetRunDurationUSec() * 1.0 / 1000000);
-        bad_status();
-      }
-      int priority = 12;
-      if ((*it)->GetErrorCount()) priority = 5;
```

Comment:
You're losing this logic ?

Reply:
That's just adding process errors based on the threads having a bad status - the threads now add errors themselves when they happen instead of doing this cleanup step at the end.